openfisca_core/formulas.py

```diff
@@ -774,15 +770,14 @@ def new_filled_column(
     elif isinstance(source_file_path, str):
         source_file_path = source_file_path.decode('utf-8')
 
-    if start_date is UnboundLocalError:
-        start_date = None if reference_column is None else reference_column.start
-    elif start_date is not None:
-        assert isinstance(start_date, datetime.date)
-
-    if stop_date is UnboundLocalError:
-        stop_date = None if reference_column is None else reference_column.end
-    elif stop_date is not None:
-        assert isinstance(stop_date, datetime.date)
+    if end is UnboundLocalError:
+        end = None if reference_column is None else reference_column.end
+    elif end is not None:
+        assert isinstance(end, str), 'Type error on {}. String expected. Found: {}'.format(name + '.end', type(end))
+        try:
+            get_datetime_date(end)
```

Comment:
I suggest to declare an `end` attribute of `date` type to Formula class. There is no desagreement about not parsing a string multiple times.